core/jobs/batch_jobs/store_profile_images_to_gcs_jobs.py

```diff
@@ -0,0 +1,338 @@
+                    object_image[1]['model_picture'])
+            )
+        )
+
+        report_mismatched_png_images_on_gcs_and_model = (
+            mismatched_png_images_on_gcs_and_model
+            | 'Report the png data' >> beam.Map(lambda data: (
+                job_run_result.JobRunResult.as_stderr(
+                    'The user having username %s, have mismatched png image on '
+                    'GCS and in the model.' % (data[0])
+                )
+            ))
+        )
+
+        total_mismatched_png_images = (
+            mismatched_png_images_on_gcs_and_model
+            | 'Total number of mismatched png images' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'TOTAL MISMATCHED PNG IMAGES'))
+        )
+
+        png_images_iterated_on_gcs = (
+            audit_png_profile_pictures
+            | 'Total number of png images iterated' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'TOTAL PNG IMAGES ITERATED ON GCS'))
+        )
+
+        # Audit webp images.
+        audit_webp_profile_pictures = (
+            users_with_valid_username
+            | 'Map with username for webp' >> beam.Map(
+                lambda model: model.username)
+            | 'Map with filename for webp' >> beam.Map(
+                lambda username: f'user/{username}/profile_picture.webp')
+            | 'Read webp files from GCS' >> gcs_io.ReadFile(
+                client=self.client, mode_is_binary=True)
+            | 'Make tuple of username and data url for webp' >> beam.Map(
+                lambda data: (
+                    data[0].split('/')[1],
+                    utils.convert_png_or_webp_binary_to_data_url(data[1], True))
+            )
+        )
+
+        username_with_profile_data_webp = (
+            username_with_profile_data
+            | 'Convert to webp base64 string' >> beam.Map(
+                lambda data: (data[0], png_base64_to_webp_base64(data[1])))
+        )
+
+        mismatched_webp_images_on_gcs_and_model = (
+            {
+                'gcs_picture': audit_webp_profile_pictures,
+                'model_picture': username_with_profile_data_webp
+            }
+            | 'Merge models for webp' >> beam.CoGroupByKey()
+            | 'Filter invalid webp images' >> beam.Filter(
+                lambda object_image: (
+                    object_image[1]['gcs_picture'] !=
+                    object_image[1]['model_picture'])
+            )
+        )
+
+        report_mismatched_webp_images_on_gcs_and_model = (
+            mismatched_webp_images_on_gcs_and_model
+            | 'Report the webp data' >> beam.Map(lambda data: (
+                job_run_result.JobRunResult.as_stderr(
+                    'The user having username %s, has incompatible webp image '
+                    'on GCS and png in the model.' % (data[0])
+                )
+            ))
+        )
```

Comment:
Hey @vojtechjelinek,
Please note that the reason why I'm not outputting the image on gcs and in the model when they are not same, is because it is going to get [TRUNCATED] and we will not be able to compare the output anyway.
For the webp part I'm converting the GCS webp image to png and then comparing them with the `profile_picture_data_url` in the model as it seems to only way to verify the audit.
Thanks

Reply:
Makes sense